Update to remove awx.awx from module names

sean-m-sullivan opened the pull request:
### What does this PR do?
This PR updates the yaml and ansible lint to their latest versions, and by extension allows us to remove awx.awx from module names and still pass ansible lint. These are then tested against a deployment of awx, and have been tested on tower.


### Is there a relevant Issue open for this?
resolves #115

sean-m-sullivan committed:
update awx integration test

sean-m-sullivan committed:
Merge pull request #46 from redhat-cop/devel

Rebase

sean-m-sullivan committed:
update

sean-m-sullivan committed:
update

sean-m-sullivan committed:
update

sean-m-sullivan committed:
update

sean-m-sullivan committed:
update

sean-m-sullivan committed:
update

sean-m-sullivan committed:
update

sean-m-sullivan committed:
update

sean-m-sullivan committed:
update

sean-m-sullivan committed:
update

sean-m-sullivan committed:
update

sean-m-sullivan committed:
update

Tompage1994 commented on changelogs/fragments/ansible_lint.yml:
```diff
@@ -0,0 +1,5 @@
+major_changes:
+  - Updated collection to use and comply with ansible-lint v5
+breaking_changes:
+  - Updated to allow use of either awx.awx or ansible.tower
```
Is it actually a breaking change? Will old playbooks fail? If so it may be worth having a major change log in the README so anybody who comes back to the collection is aware of what needs to be changed in their playbooks

sean-m-sullivan replied:
Without the opening playbook listing awx.awx, its not going to find the right modules specifcally for applications and tower notification template. 

Did a deep dive and for those that were included in regular ansible 2.9, they map it to the new awx.awx modules, and those aren't mapped as they didn't come out till later.

Tompage1994 replied:
ok, probably worth calling out in the README that the collection needs to be specified in the playbook?

sean-m-sullivan replied:
Updated the base readme under using the collection, Should we update it elsewhere?

Tompage1994 reviewed (approved):
Just one question to check but otherwise happy 👍

sean-m-sullivan committed:
update

sean-m-sullivan committed:
Merge branch 'devel' into lint_update

sean-m-sullivan committed:
update ansible-lint version to non rc, and fix lint issue

sean-m-sullivan reviewed (commented):
Reviewed all the files again, fixed a few errors, I think its ready.

sean-m-sullivan committed:
lint

sean-m-sullivan committed:
update lint rules and linting

sean-m-sullivan commented:
After rereading yaml lint rules, and seeing as warnings were not showing up easily, I've updated the lint rules to fit our criteria, and also adjusted the colon rule and several spaces before colon areas we were missing.

sean-m-sullivan committed:
updated mandatory on workflow nodes file

Tompage1994 commented on README.md:
```diff
@@ -27,6 +27,20 @@ collections:
 ## Using this collection
+The awx.awx or ansible.tower collection must be invoked in the playbook in order for ansible to pick up the correct modules to use.
+
+Otherwise it will look for for the modules only in your base installation. If there are errors complaining about "counldn't resolve module/action' this is the most likely cause.
```
`Otherwise it will look for for the modules only in your base installation. If there are errors complaining about "couldn't resolve module/action" this is the most likely cause.`

sean-m-sullivan replied:
Thought you were just highlighting missed what you pointed out later.

Tompage1994 reviewed (changes requested):
Just a single minor typo in the readme. Probably best to make sure the main readme is as good as possible. After fixing that Im completely happy for this to be merged

sean-m-sullivan committed:
update

Tompage1994 commented on README.md:
```diff
@@ -29,7 +29,7 @@ collections:
 The awx.awx or ansible.tower collection must be invoked in the playbook in order for ansible to pick up the correct modules to use.
 
-Otherwise it will look for for the modules only in your base installation. If there are errors complaining about "counldn't resolve module/action' this is the most likely cause.
+Otherwise it will look for the modules only in your base installation. If there are errors complaining about "counldn't resolve module/action' this is the most likely cause.
```
That wasn't even the typo I spotted 😂

Could you replace `counldn't` with `couldn't` and then the final single quote should be a double quote

sean-m-sullivan replied:
Whoops on it.

sean-m-sullivan committed:
typos

Tompage1994 reviewed (approved):
Looks good now
